src/deadline/job_attachments/job_attachments_s3_bucket_lister.py

```
@@ -186,54 +153,93 @@ def list_job_attachments_with_prefixes(
 
 class S3InventoryLister(JobAttachmentsS3BucketLister):
     """
-    Lister that uses an S3 Inventory manifest to list objects from S3.
+    Object fetcher that uses an S3 Inventory manifest to list objects from S3.
 
     See for example data:
         https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-inventory.html
 
+    Warning:
+        This implementation loads the entire manifest file into memory during initialization. Large
+        manifest files may exhaust memory resources.
```

Could we fail fast by checking the file size and local memory available? That'd help with

1. give specific and actionable error message such as cleaning local storage before proceeding 
2. avoid failed downloads if file is too big
3. save time & bandwidth by skipping unnecessary transfers.

We can query its metadata using HEAD requests (HeadObject) to retrieve the same HTTP headers without retrieving the body of the object.

---

Yes, good idea. Will change.